### PR TITLE
DM-43445: Camera load in Prompt Processing assumes a single unbounded collection

### PR DESCRIPTION
**Pull request by kfindeisen**
This PR modifies the code for loading the initial `camera` dataset to get it from `<instrument>/calib` where possible, keeping the old behavior of `<instrument>/calib/unbounded` only as a fallback for older repos.

### CODE REVIEW

File: python/activator/middleware_interface.py [hunk 1]
```diff
@@ -270,13 +270,20 @@ def __init__(self, central_butler: Butler, image_bucket: str, visit: FannedOutVi
...
+                collections=self.instrument.makeCalibrationCollectionName(),
+            )
+        except lsst.daf.butler.DatasetNotFoundError:
+            # Repos that don't allow retrieval of camera from <instrument>/calibs
+            # have <instrument>/calibs/unbounded.
+            self.camera = self.central_butler.get(
+                "camera",
+                instrument=self.instrument.getName(),
+                collections=self.instrument.makeUnboundedCalibrationRunName(),
+            )
```
Comment (TallJimbo) [inline python/activator/middleware_interface.py:286]: I think this is a possible alternative:
```
camera_ref = self.central_butler.find_dataset(
    "camera",
    instrument=self.instrument.getName(),
    collections=self.instrument.makeCalibrationCollectionName(),
    timespan=Timespan(begin=None, end=None)
)
self.camera = butler.get(camera_ref)
```
it's a shame `Butler.get` doesn't accept `timespan` and forward that down to its own `find_dataset` call directly.

Your original looks fine, but if what I've written works, it'll avoid dependence on one of our shakier convention.

**Review by TallJimbo — APPROVED** [review]

File: python/activator/middleware_interface.py (hunk 1 above)
Comment (timj) [reply python/activator/middleware_interface.py:286]: I will look at forwarding timespan in [DM-43499](https://rubinobs.atlassian.net/browse/DM-43499).

Comment (kfindeisen) [reply python/activator/middleware_interface.py:286]: I'd like to give it a shot, but the "live" tests have turned up serious problems that will keep me busy up until the code freeze. I'll add a TODO comment and try your suggestion as part of [DM-42825](https://rubinobs.atlassian.net/issues/DM-42825), which already involves expanding our use of `find_dataset`.

Comment (timj) [reply python/activator/middleware_interface.py:287]: This support was added in lsst/daf_butler#982

**Pull request merged**